### DISCUSSION TITLE
Downloading data through OpenDataLab and putting the file in data/cache/ does not automatically prepare the data.

### DISCUSSION THREAD

**Discussion by htaoruan** [open-mmlab/mmocr#1835] [General]
When I download SynthText.tar.gz data through OpenDataLab and put the file in data/cache/, the data is not prepared automatically. 
python tools/dataset_converters/prepare_dataset.py synthtext --task textrecog --overwrite-cfg

tools/dataset_converters/prepare_dataset.py:133: UserWarning: synthtext is not supported yet. Please check dataset zoo for supported datasets.
  warnings.warn(f'{dataset} is not supported yet. Please check '

**Comment by gaotongxiao** [comment]
Switch to `dev-1.x` branch and try again.

**Reply by htaoruan** [reply]
Okay, I'll try it right away

**Reply by htaoruan** [reply]
hello,Modify save_name='SynthText.tar.gz' in /mmocr-dev-1.x/dataset_zoo/synthtext/textdet.py.
lib/python3.8/urllib/request.py", line 1425, in unknown_open
    raise URLError('unknown url type: %s' % type)
urllib.error.URLError: <urlopen error unknown url type: magnet>

**Reply by htaoruan** [reply]
The original is SynthText.zip

**Reply by gaotongxiao** [reply]
Try setting 'md5' as None

**Reply by htaoruan** [reply]
Obtaining train Dataset...
openmmlab/mmocr-new/mmocr-1.x/mmocr/utils/fileio.py:78: UserWarning: MD5 is None, skip the integrity check.
  warnings.warn('MD5 is None, skip the integrity check.')
Extracting: SynthText.tar.gz
Start to download subset_textrecog_train.json...
If you stuck here for a long time, please check your network.
Downloading subset_textrecog_train.json: 100.00%
Start to download alphanumeric_textrecog_train.json...
If you stuck here for a long time, please check your network.
Downloading alphanumeric_textrecog_train.json: 100.00%
Gathering train Dataset...
Parsing train Images and Annotations...

KeyError: 'SynthTextAnnParser is not in the data parser registry. Please check whether the value of `SynthTextAnnParser` is correct or it was registered as expected. More details can be found at https://mmengine.readthedocs.io/en/latest/advanced_tutorials/config.html#import-the-custom-module'

**Reply by gaotongxiao** [reply]
Did you install MMOCR via MIM but now trying to use its source code? Try to add `PYTHONPATH=.` before your command.

**Reply by htaoruan** [reply]
Thank you for your previous help.
now   FileNotFoundError: [Errno 2] No such file or directory: 'data/synthtext/annotations/gt. mat' is this file required? How should we avoid it?

**Reply by gaotongxiao** [reply]
Yes, all the annotations are saved in this mat file. You may manually move `gt.mat` to the path it gives. The file should be inside the `tar.gz' file.
